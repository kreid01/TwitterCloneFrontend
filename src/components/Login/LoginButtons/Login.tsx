import React from "react";
import { Link } from "react-router-dom";

interface Props {
  openLoginPage: (option: string) => void;
}

export const Login: React.FC<Props> = ({ openLoginPage }) => {
  return (
    <div className=" h-14 bg-blue-500 fixed bottom-0 w-screen flex justify-center">
      <Link
        to="/login"
        onClick={() => openLoginPage("login")}
        className="login-button"
      >
        Log In
      </Link>
      <button onClick={() => openLoginPage("signup")} className="signup-button">
        Sign Up
      </button>
    </div>
  );
};
