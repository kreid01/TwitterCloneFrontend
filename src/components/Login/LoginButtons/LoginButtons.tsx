import React from "react";
import { Link } from "react-router-dom";

interface Props {
  openLoginPage: (option: string) => void;
}

export const LoginButtons: React.FC<Props> = ({ openLoginPage }) => {
  return (
    <div className="h-14 ml-14 bg-blue-500 fixed bottom-0  w-[84vw] md:w-[75vw] lg:w-[70vw] lg:mb-2  lg:ml-1 flex justify-center">
      <Link
        to="/login"
        data-testid="loginButton"
        onClick={() => openLoginPage("login")}
        className="login-button"
      >
        Log In
      </Link>
      <button
        data-testid="signupButton"
        onClick={() => openLoginPage("signup")}
        className="signup-button"
      >
        Sign Up
      </button>
    </div>
  );
};
