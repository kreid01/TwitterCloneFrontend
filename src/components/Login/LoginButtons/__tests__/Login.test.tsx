/* eslint-disable testing-library/prefer-screen-queries */
import React from "react";
import { LoginButtons } from "../LoginButtons";

import { screen, render, cleanup, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";

import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";

const mockFn = jest.fn();

afterEach(cleanup);
it("render without crashing", () => {
  render(
    <BrowserRouter>
      <LoginButtons openLoginPage={mockFn} />
    </BrowserRouter>
  );
});

it("calls handleTweeet function when clicked", () => {
  render(
    <BrowserRouter>
      <LoginButtons openLoginPage={mockFn} />
    </BrowserRouter>
  );
  fireEvent.click(screen.getByTestId("signupButton"));
  expect(mockFn).toHaveBeenCalled();
});

it("matched snapshot", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <LoginButtons openLoginPage={mockFn} />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("calls takes user to login", () => {
  render(
    <BrowserRouter>
      <LoginButtons openLoginPage={mockFn} />
    </BrowserRouter>
  );
  fireEvent.click(screen.getByTestId("loginButton"));
  expect(window.location.pathname).toBe("/login");
});
