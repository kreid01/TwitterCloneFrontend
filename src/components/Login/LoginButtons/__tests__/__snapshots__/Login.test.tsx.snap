// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matched snapshot 1`] = `
<div
  className=" h-14 bg-blue-500 fixed bottom-0 w-screen flex justify-center"
>
  <a
    className="login-button"
    data-testid="loginButton"
    href="/login"
    onClick={[Function]}
  >
    Log In
  </a>
  <button
    className="signup-button"
    data-testid="signupButton"
    onClick={[Function]}
  >
    Sign Up
  </button>
</div>
`;
